tests/v2/test_wakunode.nim

```diff
@@ -144,3 +144,40 @@ procSuite "WakuNode":
       (await completionFut.withTimeout(5.seconds)) == true
     await node1.stop()
     await node2.stop()
+
+  asyncTest "Filter protocol returns expected message":
+    let
+      nodeKey1 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node1 = WakuNode.init(nodeKey1, ValidIpAddress.init("0.0.0.0"),
+        Port(60000))
+      nodeKey2 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node2 = WakuNode.init(nodeKey2, ValidIpAddress.init("0.0.0.0"),
+        Port(60002))
+      contentTopic = "foobar"
+      message = WakuMessage(payload: "hello world".toBytes(), contentTopic: contentTopic)
+
+    var completionFut = newFuture[bool]()
+
+    await node1.start()
+    await node2.start()
+
+    node1.wakuFilter.setPeer(node2.peerInfo)
```

Comment:
Out of scope of PR but would be good with an enhancement issue here to set multiple peers. (Same goes for store protocol)

Reply:
Yeah, there is a todo for this.

Reply:
Where?

Reply:
https://github.com/status-im/nim-waku/blob/feature/mount-filter-node-api/waku/protocol/v2/waku_filter.nim#L129 Will be turned into an issue too.